### ISSUE TITLE
Athena/Trino dialect fails to parse `json_object` statement

### ISSUE THREAD

**Issue by lehela** [sqlfluff/sqlfluff#5558]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

sqlfluff fails to parse a variant of the `json_object` statement for the dialects `athena` and `trino`:
```sql
SELECT json_object('key1' : 1, 'key2' : true)
```
Observe, that other valid `json_object` variants are parsed successfully, e.g.
```sql
SELECT json_object(KEY 'key1' VALUE 1, KEY 'key2' VALUE true);
SELECT json_object('key1' VALUE 1, 'key2' VALUE true);
```
> The statements above are taken from the corresponding [Trino documentation](https://trino.io/docs/current/functions/json.html#json-object), but they also work in Athena.

### Expected Behaviour

sqlfluff should parse the statement successfully.

### Observed Behaviour

sqlfluff reports a parsing violation:
```
==== parsing violations ====
L:   1 | P:  27 |  PRS | Line 1, Position 27: Found unparsable section: ": 1, 'key2' :
                       | true"
WARNING: Parsing errors found and dialect is set to 'athena'. Have you configured your dialect correctly?
```

### How to reproduce

Create sql file `test.sql`
```sql 
-- test.sql
SELECT json_object('key1' : 1, 'key2' : true)
```

Parse with `sqlfluff` using the dialect `trino`
```bash
sqlfluff parse test.sql --dialect trino
```

Alternatively, parse with `sqlfluff` using the dialect `athena`
```bash
sqlfluff parse test.sql --dialect athena
```

### Dialect

athena
trino

### Version

sqlfluff, version 2.3.5
Python 3.8.10

### Configuration

[sqlfluff]
dialect = athena
max_line_length = 150
warnings = RF02, RF04, AL03, ST03

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
